### PR TITLE
8313711: Cherry-pick WebKit 616.1 stabilization fixes

### PR DESCRIPTION
**Pull request by HimaBinduMeda**
Cherry-picked commits from webkitgtk-2.40.5.
Verified build on windows, mac and linux. Sanity testing looks fine. No issues seen

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8313711](https://bugs.openjdk.org/browse/JDK-8313711): Cherry-pick WebKit 616.1 stabilization fixes (**Bug** - P3)

### Reviewers
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Joeri Sykora](https://openjdk.org/census#sykora) (@tiainen - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1196/head:pull/1196` \
`$ git checkout pull/1196`

Update a local copy of the PR: \
`$ git checkout pull/1196` \
`$ git pull https://git.openjdk.org/jfx.git pull/1196/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1196`

View PR using the GUI difftool: \
`$ git pr show -t 1196`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1196.diff">https://git.openjdk.org/jfx/pull/1196.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1196#issuecomment-1664421277)

### CODE REVIEW

**Comment by kevinrushforth** [comment]
I changed the JBS title to match the pattern that we use for cherry-picking stabilization fixes.

/reviewers 2

**Comment by kevinrushforth** [comment]
Note that this also contains cherry-picked commits from webkitgtk-2.40.4, since the WebKit 616.1 update was synced up through webkitgtk-2.40.3.

Reviewers: @kevinrushforth + @tiainen or @johanvos

File: modules/javafx.web/src/main/native/Source/JavaScriptCore/b3/B3DuplicateTails.cpp [hunk 1]
```diff
@@ -77,6 +77,15 @@ class DuplicateTails {
...
+            bool canCopyBlock = true;
+            for (Value* value : *block) {
+                if (value->kind().isCloningForbidden()) {
+                    canCopyBlock = false;
+                    break;
+                }
+            }
+
+            if (canCopyBlock)
             candidates.add(block);
```
Comment (kevinrushforth) [inline modules/javafx.web/src/main/native/Source/JavaScriptCore/b3/B3DuplicateTails.cpp:89]: The indentation looks off here.

**Review by kevinrushforth — APPROVED** [review]
Looks good. Testing is green.

File: modules/javafx.web/src/main/native/Source/JavaScriptCore/b3/B3DuplicateTails.cpp (hunk 1 above)
Comment (kevinrushforth) [reply modules/javafx.web/src/main/native/Source/JavaScriptCore/b3/B3DuplicateTails.cpp:89]: I'll reapprove if you choose to fix this.

**Review by tiainen — APPROVED** [review]
All tests ran fine here too.

File: modules/javafx.web/src/main/native/Source/JavaScriptCore/b3/B3DuplicateTails.cpp (hunk 1 above)
Comment (HimaBinduMeda) [reply modules/javafx.web/src/main/native/Source/JavaScriptCore/b3/B3DuplicateTails.cpp]: Indentation corrected

**Review by kevinrushforth — APPROVED** [review]

**Pull request closed without merging**